components/arm/yahboom/dofbot.go

```diff
@@ -293,14 +294,14 @@ func (a *Dofbot) readJointInLock(ctx context.Context, joint int) (float64, error
 
 	time.Sleep(3 * time.Millisecond)
 
-	res, err := a.handle.ReadWordData(ctx, reg)
+	bytes, err := a.handle.ReadBlockData(ctx, reg, 2)
 	if err != nil {
 		return 0, fmt.Errorf("error reading joint %v from register %v: %w", joint, reg, err)
 	}
 
 	time.Sleep(3 * time.Millisecond)
 
-	res = (res >> 8 & 0xff) | (res << 8 & 0xff00)
+	res := binary.BigEndian.Uint16(bytes)
```

Comment:
[test-req] let's test on the dofbot. But seems fine on a cursory look.

Reply:
The dofbot doesn't seem to work very well, but it performs roughly the same on both the main branch and this branch. I'd be happy to show you what I've got.

Reply:
I showed Rand: we filed two Jira tickets to clean up some of the dofbot mess, though we both think this PR has not made anything worse than it already is.